### ISSUE TITLE
arc-versions.sh fails to checkout gcc/arc-4.8-dev branch

### ISSUE THREAD

**Issue by nandub** [foss-for-synopsys-dwc-arc-processors/gcc#31]
When script tries to checkout arc-4.8-dev branch fails with "The remote end hung up unexpectedly"

```
ARC_GNU=../ ./arc-versions.sh --no-uclibc
Checking out branch/tag arc-1.0-dev of cgen
  fetching branches
  fetching tags
  checking out arc-1.0-dev
Already on 'arc-1.0-dev'
Your branch is up-to-date with 'origin/arc-1.0-dev'.
  pulling latest version
Already up-to-date.
Checking out branch/tag arc-2.23-dev of binutils
  fetching branches
  fetching tags
  checking out arc-2.23-dev
Already on 'arc-2.23-dev'
Your branch is up-to-date with 'origin/arc-2.23-dev'.
  pulling latest version
Already up-to-date.
Checking out branch/tag arc-4.8-dev of gcc
  fetching branches
  fetching tags
remote: Counting objects: 1204870, done.
remote: Compressing objects: 100% (199708/199708), done.
error: RPC failed; result=56, HTTP code = 200.96 MiB | 311.00 KiB/s   
fatal: The remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed
```

I'm using Mac OS X Mountain Lion

**Comment by nandub** [comment]
It is also failing with the linux repo.  I think github is having problem with big repositories.

**Comment by anthony-kolesov** [comment]
What protocol are you using? git, http, https?
You can see it with:

```
[gcc] $ git remote -v
```

git protocol is prefered, you can change in existing repository:

```
[gcc] $ git remote set-url origin git://github.com/foss-for-synopsys-dwc-arc-processors/gcc.git
```

Then try to `git fetch` manually. Then `git fetch -t`, because `arc-versions.sh` script also does that.

If you are already on git protocol, then maybe try http instead. It looks like something with Git for Mac, I don't remember anything like that on Linux. What Git version are you using?

**Comment by mischajonker** [comment]
Hi Fernando,

Maybe these links are of use?

http://stackoverflow.com/questions/6842687/the-remote-end-hung-up-unexpectedly-while-git-cloning
http://codeplex.codeplex.com/workitem/26133
https://bugs.launchpad.net/ubuntu/+source/gnutls26/+bug/1111882

Depending on what you would like to do, there are other solutions available, such as:
- use Buildroot (for uclibc toolchain, Linux; NOT for elf32)
- use prebuilt toolchain
- use the github source download interface: https://github.com/foss-for-synopsys-dwc-arc-processors/gcc/tarball/(commit-id) will give you a .tar.gz of the commit ID requested

**Comment by nandub** [comment]
Mischa,

Thanks, is making sense now I was using https instead of git.  After a third try it works but now when I run the following command I get this error:

```
./build-all.sh --no-uclibc --install-dir /opt/arc-elf32
uname: illegal option -- o
usage: uname [-amnprsv]
```

It looks like uname on this mac does not have the `-o` option :(

REASON: I was trying to create an ARC toolchain from scratch using my mac.  After I searched for a while I found this github repository that I can use to build a toolchain?

I have a wireless access point with the following specs:

```
uname -a

Linux soc1 2.6.30 #2 Mon Jun 24 10:56:44 PDT 2013 arc unknown

cat /proc/cpuinfo

Processor Family: ARC 700, Core ID: 0
CPU speed : 400.00 Mhz
Timers:     TIMER1 TIMER0 
Interrupt Vect Base:    0x8800ec00 
Peripheral Base:    0x0 
Data UNCACHED Base (I/O): start 0xc0 Sz, 1024 MB 
Bogo MIPS :     265.42
ARC700 MMU Ver [2]
   uDTLB 8 entr, uITLB 4 entr, JTLB 128 entry/way, 2 ways
Detected I-cache : 
  Type=2 way set-assoc, Line length=32, Size=16K (enabled)
Detected D-cache : 
  Type=4 way set-assoc, Line length=32, Size=16K (enabled)
Extensions:
Multiplier: 32x32 with ANY Result Reg   MAC Multiplier: Dual 16 x 16 and 32 x 16
CRC  Instrn: NA,   SWAP Instrn: Present   NORM Instrn: Present
Min-Max Instrn: Present,   Barrel Shift Rotate Instrn: Present
Ext Arith Instrn: Present
```

**Comment by mischajonker** [comment]
Hi Fernando,

The toolchains here on github are the most recent ones for the ARC processor family. These work together with the Linux kernel that is available on kernel.org.

Your Wifi router has a kernel from before that age (< Linux 3.9), and is not compatible with the toolchain on github. So, depending on what you want to do with the toolchain, you may need an older toolchain.
1. If you just want to compile some user-space programs to run on your router, you need the old toolchain.
2. If you want to port a recent Linux kernel (>= 3.9) to your router and rebuild the root file system etc from scratch, then you need the new toolchain.
3. If you want to build an old (< v3.9) kernel, you can use the old toolchain as well.

I'm guessing you don't want to do 2. so, let me point you to the old toolchain:
http://sourceforge.net/projects/arc-linux/files/ARC%20GNU%20Toolchain/ARC%20Rel%202.3/

Here's some more info on these different ABI's:
https://github.com/foss-for-synopsys-dwc-arc-processors/linux/wiki/ARC-Linux-Syscall-ABI-Compatibility

BTW: we haven't tested these toolchains on a Mac. It may or may not work. If you experience problems, I recommend to use VirtualBox with Ubuntu. For these old toolchains you may even need to use a 32 bit operating system!

**Comment by claziss** [comment]
Hi Fernando,

I have not installed the uClibc toolchain on Mac, but for elf toolchain, you the three libs (gmp, mpfr, mpc) installed on your system. Also you need to tweak the build scripts to consider the paths where these libs are installed. On top of this you need for sure Xcode installed, and, maybe, mac ports.

//Claudiu

**Comment by nandub** [comment]
Mischa,

Thanks for the great response and information
- If I were to do # 2, do you know if the Wifi AP from specs above will support the new kernel?  
- What batch file scripts should I look at to make changes if needed? or should I use your buildroot fork to build everything?

**Comment by vineetgarc** [comment]
Hi Fernando,

I'm curious what actual device is this. I've been maintaining ARC Linux kernel for a bit and I assumed I knew all customers/ODMs using Linux on ARC processors. Will you be able to share information about this box. Perhaps I could use it my home office to eat my own dog food ;-)

**Comment by nandub** [comment]
Actually this is a VAP2500 WAP use exclusively by AT&T U-verse, to make the story short we decided it to cancel the service and when I returned the equipment back to AT&T via UPS, the personnel told me they throw away every thing except the U-verse gateway and the DVRs.  The cables and the WAP is thrown away so they gave me the choice to keep the WAP.

This particular box is vulnerable to the [PHP-CGI bug](http://www.pentesterlab.com/exercises/cve-2012-1823) which is the reason why I'm able to get inside the box.
